### PR TITLE
Updates corresponding to improvements to the consumer libraries

### PR DESCRIPTION
**Pull request by ntl**

### CODE REVIEW

File: user-guide/consumers.md [hunk 1]
```diff
@@ -200,14 +200,35 @@ Consumer.start('account:command')
...
+SomeConsumer.start('someCategory', condition: "hash64(stream_name) % 2 = 0")
+SomeConsumer.start('someCategory', condition: "hash64(stream_name) % 2 = 1")
+```
+
+In the above example, two instances of the same consumer have been started, each consuming different entity streams within the category stream `someCategory`.
+
+::: danger
+Usage of this feature should be treated with caution, as it can cause messages to be read by consumers in a different order in which they were written. Idempotence techniques which rely on processing messages in-order therefore may not work in conjuction with `conditions`. Therefore, it is recommended that this feature be used with care, and only in cases where the ramifications are fully understood.
+:::
+
```
Comment (ntl) [inline user-guide/consumers.md:222]: The example isn't super clear, but I couldn't think of a simpler example that was "safe" to use in applicative code. I wonder if we should omit the example altogether.

Comment (ntl) [reply user-guide/consumers.md:222]: Actually, I'm wondering if correlation stream category would make more sense here. It's a common use case for pub/sub, and the reason that is most likely to introduce users to this feature.

Comment (ntl) [reply user-guide/consumers.md:222]: **Update**: I have reworked the example to be about filtering out messages that don't correlate to a particular entity.

Also, there should probably be a link to the correlation section of the [metadata](https://github.com/eventide-project/docs/blob/master/user-guide/messages-and-message-data/metadata.md) document in here, but I'm unsure how to link documentation in this project.

File: user-guide/consumers.md [hunk 2]
```diff
@@ -91,8 +91,8 @@ self.start(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, posit
 | poll_interval_milliseconds | The frequency, in milliseconds, with which the consumer polls the message store for new messages | Integer |
 | batch_size | The number of messages to retrieve in each batch fetched from the message store | Integer |
 | position_update_interval | The frequency with which progress that the consumer has made through the input stream is recorded by the [position store](#position-store) | Integer |
-| position_store | Position store that records the consumer's progress through the input stream | Consumer::Postgres::PositionStore |
-| session | An existing [session](./session.md) object to use, rather than allowing the consumer to create a new session | MessageStore::Postgres::Session |
+| settings | Settings that can configure a [session](./session.md) object for the consumer to use, rather than the default settings read from `settings/message_store_postgres.json` | MessageStore::Postgres::Settings |
+| condition | SQL condition fragment that constrains the messages of the stream that are read |
```
Comment (sbellware) [inline user-guide/consumers.md:95]: Missing parameter type for condition

File: user-guide/consumers.md [hunk 3]
```diff
@@ -80,7 +80,7 @@ SomeConsumer.start('someStream')
 A consumer's `start` method offers a number of parameters to control its timing.
 
 ``` ruby
-self.start(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, position_update_interval: 100, position_store: nil, session: nil)
+self.start(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, position_update_interval: 100, settings: nil, condition: nil)
```
Comment (sbellware) [inline user-guide/consumers.md:83]: This seems like it should be documented as session coming at the end of the list of parameters

File: user-guide/consumers.md [hunk 4]
```diff
@@ -200,14 +200,34 @@ Consumer.start('account:command')
 
 In the above example, the consumer's position stream would be `account:command-someConsumer`.
 
+Consumers can also be assigned an identifier when they are started. If an identifier macro is also declared on the consumer class, the one given when starting the consumer is chosen.
+
```
Comment (sbellware) [inline user-guide/consumers.md:204]: Maybe should be: the one given when starting the consumer takes precedence

File: user-guide/consumers.md (hunk 1 above)
Comment (sbellware) [reply user-guide/consumers.md:222]: 👍

File: user-guide/consumers.md [hunk 5]
```diff
@@ -200,14 +200,34 @@ Consumer.start('account:command')
...
 ## Constructing Consumers
 
 In general, it's not necessary to construct a consumer. The general use case of a consumer is to invoke its `start` method.
 
 A consumer can be constructed with its `build` method.
 
 ``` ruby
-self.build(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, position_update_interval: 100, position_store: nil, session: nil)
+self.build(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, position_update_interval: 100, settings: nil, condition: nil)
```
Comment (sbellware) [inline user-guide/consumers.md:230]: Again, I think settings should be the last argument in the documentation (if not the signature definition)

File: user-guide/consumers.md [hunk 6]
```diff
@@ -218,5 +238,5 @@ self.build(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, posit
 | poll_interval_milliseconds | The frequency, in milliseconds, with which the consumer polls the message store for new messages | Integer |
 | batch_size | The number of messages to retrieve in each batch fetched from the message store | Integer |
 | position_update_interval | The frequency with which progress that the consumer has made through the input stream is recorded by the [position store](#position-store) | Integer |
-| position_store | Position store that records the consumer's progress through the input stream | Consumer::Postgres::PositionStore |
-| session | An existing [session](./session.md) object to use, rather than allowing the consumer to create a new session | MessageStore::Postgres::Session |
+| settings | Settings that can configure a [session](./session.md) object for the consumer to use, rather than the default settings read from `settings/message_store_postgres.json` | MessageStore::Postgres::Settings |
+| condition | SQL condition fragment that constrains the messages of the stream that are read |
```
Comment (sbellware) [inline user-guide/consumers.md:242]: Missing parameter type of the condition

**Review by sbellware — CHANGES_REQUESTED** [review]

**Pull request merged**